tests/test_huggingfacemodel.py

```diff
@@ -0,0 +1,140 @@
+# OBSS SAHI Tool
+# Code written by Fatih C Akyon, 2020.
+
+import unittest
+
+import pybboxes.functional as pbf
+
+from sahi.utils.cv import read_image
+
+MODEL_DEVICE = "cpu"
+CONFIDENCE_THRESHOLD = 0.3
+IMAGE_SIZE = 320
+TEST_MODEL_PATH = "hustvl/yolos-tiny"
+
+
+class TestYolov5DetectionModel(unittest.TestCase):
+    def test_load_model(self):
+        from sahi.model import HuggingfaceDetectionModel
+
+        yolov5_detection_model = HuggingfaceDetectionModel(
+            model_path=TEST_MODEL_PATH,
+            confidence_threshold=CONFIDENCE_THRESHOLD,
+            device=MODEL_DEVICE,
+            category_remapping=None,
+            load_at_init=True,
+        )
+
+        self.assertNotEqual(yolov5_detection_model.model, None)
+
+    def test_set_model(self):
+        from transformers import AutoModelForObjectDetection
+
+        from sahi.model import Yolov5DetectionModel
+
+        huggingface_model = AutoModelForObjectDetection.from_pretrained(TEST_MODEL_PATH)
+
+        yolov5_detection_model = Yolov5DetectionModel(
+            model=huggingface_model,
+            confidence_threshold=CONFIDENCE_THRESHOLD,
+            device=MODEL_DEVICE,
+            category_remapping=None,
+            load_at_init=True,
+        )
+
+        self.assertNotEqual(yolov5_detection_model.model, None)
+
+    def test_perform_inference(self):
```

Comment:
Maybe we can an additional test as `test_automodel_prediction` to test HuggingfaceDetectionModel inference via AutoModel class.

Reply:
Automodel is only a factory class for loading the desired detection model, so I didn't quite get it how we can test it. Can you give an example ?

Reply:
What I am talking about is something like this: https://github.com/obss/sahi/blob/5320f48a1c8f1b5e4d2168c44ae3fb7e94511139/tests/test_predict.py#L121

You can consider it as an integration test.

It is not compulsory though, we can skip it.

Reply:
@fcakyon Oh okay, I can add it. I've mistaken as transformers' automodel.

Reply:
@fcakyon all tests are added in `test_predict.py` for HF model.